x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/InferNonNullAggConstraint.java

```diff
@@ -56,6 +57,12 @@ protected LogicalPlan rule(Aggregate aggregate, LocalLogicalOptimizerContext con
                 Expression field = af.field();
                 // ignore literals (e.g. COUNT(1))
                 // make sure the field exists at the source and is indexed (not runtime)
+
+                if (af instanceof AllFirst) {
```

Comment:
:+1:


Not a fan of this, but we'll be less hacky when we have more aggs like this.